_kbarticlespages/repository-best-practices.md

```diff
@@ -0,0 +1,67 @@
+
+## **Public vs Private repositories**
+
+A repository does *not need* to be public in order for Pages to be able to build the website, both public and private repositories are acceptable. When it comes to Pages there are certainly some unique considerations for each.
+
+ #### **Public**
+
+##### **Pros**
+  - Easy for Pages engineers to access the codebase and provide support/debug
+  - Transparency/High visibility
+  - Able to utilize CodeQL free of charge
+  - Seamless OpenSSF Allstar integration
+  - Collaboration with outside contributors
+  - Compliance with open source and transparency initiatives per agency/   program 
+
+##### **Cons**
+  - Any sensitive information inadvertently hardcoded is immediately visible 
+
+#### **Private**
+
+##### **Pros**
+  - Restrict access to authorized collaborators and contributors (for any support issues a Pages engineer will request to be added as a collaborator to the repository if necessary)
+  - Risk mitigation
+  - Test/Make changes to a website without exposing commits to the public
+  - Safeguard against any potential leaks of sensitive information
+
+##### **Cons**
+  - Not able to utilize CodeQL for free
+  - Potentially out of compliance with any agency/program transparency initiative
+
+## **User Permissions**
+
+It is important to designate admin/owner access of the repository to an individual(s) who is either the persistent site owner or a federal employee with the “manager” role within the Pages organization. It is strongly recommended to not grant admin/owner privileges of the repository to any contractors or other temporary employees. This is such due to certain instances where actions will need to be taken by either Pages engineers or customers where it will be necessary to have an active user with admin access to a website's repository. Additionally when adding contributors/collaborators to a private repository it should mirror the member count that your organization has within Pages. Please note that removing an organization member from an organization within Pages will **not** automatically remove them from the repository within GitHub. They will still retain whatever read/write access they have in GitHub until explicitly removed.
+
+
+## **Security**
+
+### Dependabot
+
+Dependabot is a useful and highly configurable automated dependency management tool which assists developers with keeping third-party dependencies up-to-date with their latest version. It can be extremely beneficial to smaller agencies whose developer teams are limited or in some cases a single developer by reducing maintenance overhead. It scans the repository for outdated dependencies and generates pull requests against whatever branch it is enabled on to update them. Check out our [knowledge base article on Dependabot](https://cloud.gov/pages/knowledge-base/dependabot-with-pages/#configuring-dependabot) to view how to enable and configure it for your repository.
```

Comment:
```suggestion
Dependabot is a useful and highly configurable automated dependency management tool which assists developers with keeping third-party dependencies up-to-date with their latest version. This can benefit agencies with smaller developer teams by reducing maintenance overhead. Dependabot scans the repository for outdated dependencies and generates pull requests against whatever branch it is enabled on to update them. Check out [our knowledge base article on Dependabot](https://cloud.gov/pages/knowledge-base/dependabot-with-pages/#configuring-dependabot) to learn how to enable and configure it for your repository.
```